cmd/ceremony/key.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/letsencrypt/boulder/pkcs11helpers"
+	"github.com/miekg/pkcs11"
+)
+
+type generateArgs struct {
+	mechanism    []*pkcs11.Mechanism
+	privateAttrs []*pkcs11.Attribute
+	publicAttrs  []*pkcs11.Attribute
+}
+
+func getRandomBytes(ctx pkcs11helpers.PKCtx, session pkcs11.SessionHandle) ([]byte, error) {
+	r, err := ctx.GenerateRandom(session, 4)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+const (
+	rsaModLen = 2048
+	rsaExp    = 65537
+)
+
+type keyInfo struct {
```

Comment:
Please add godoc for this. Does this represent a public key, or a private key, or a pair? On or off the HSM? Can `key` be a more specific interface, like `crypto.Signer`? What kind of ID is `id`? What structure is encoded in the DER?